tests/php/Extension/FluentSiteTreeExtensionTest.php

```diff
@@ -334,140 +332,94 @@ function (FluentState $newState) use ($localeCode, $fixture, $expected) {
-
-        FluentState::singleton()->withState(function (FluentState $newState) {
-            $newState
-                ->setLocale('de_DE')
-                ->setIsDomainMode(false)
-                ->setIsFrontend(false);
-
-            $page = Page::get()->filter('URLSegment', 'home')->first();
-
-            $this->assertNotNull($page);
-        });
-    }
-
-    public function testHomeVisibleInCMSOneConfigAny()
-    {
-        Config::modify()->set(DataObject::class, 'cms_localisation_required', FluentExtension::INHERITANCE_MODE_ANY);
-        Config::modify()->set(DataObject::class, 'frontend_publish_required', FluentExtension::INHERITANCE_MODE_EXACT);
-
-        FluentState::singleton()->withState(function (FluentState $newState) {
-            $newState
+    /**
+     * @param string $cmsInheritanceMode
+     * @param string $frontendInheritanceMode
+     * @param bool $frontendContext
+     * @param bool $expected
+     * @return void
+     * @dataProvider localeInheritanceModeCasesProvider
+     */
+    public function testLocaleInheritanceMode(
+        string $cmsInheritanceMode,
+        string $frontendInheritanceMode,
+        bool $frontendContext,
+        bool $expected
+    ): void {
+        Page::config()
+            ->set('cms_localisation_required', $cmsInheritanceMode)
+            ->set('frontend_publish_required', $frontendInheritanceMode);
+
+        FluentState::singleton()->withState(function (FluentState $state) use ($frontendContext, $expected) {
+            $state
                 ->setLocale('de_DE')
                 ->setIsDomainMode(false)
-                ->setIsFrontend(false);
+                ->setIsFrontend($frontendContext);
 
-            $page = Page::get()->filter('URLSegment', 'home')->first();
+            $page = Page::get()->find('URLSegment', 'home');
 
-            $this->assertNotNull($page);
-        });
-    }
-
-    public function testHomeNotVisibleInCMSBothConfigExact()
-    {
-        Config::modify()->set(DataObject::class, 'cms_localisation_required', FluentExtension::INHERITANCE_MODE_EXACT);
-        Config::modify()->set(DataObject::class, 'frontend_publish_required', FluentExtension::INHERITANCE_MODE_EXACT);
+            if ($expected) {
+                $this->assertNotNull($page, 'We expect the page model to have content available');
 
-        FluentState::singleton()->withState(function (FluentState $newState) {
-            $newState
-                ->setLocale('de_DE')
-                ->setIsDomainMode(false)
-                ->setIsFrontend(false);
-
-            $page = Page::get()->filter('URLSegment', 'home')->first();
+                return;
+            }
 
-            $this->assertNull($page);
+            $this->assertNull($page, 'We expect the page model to not have any content available');
         });
     }
 
-    public function testHomeNotVisibleInCMSOneConfigExact()
+    public function localeInheritanceModeCasesProvider(): array
```

Comment:
```suggestion
    public function provideLocaleInheritanceMode(): array
```
As per [the coding standards](https://docs.silverstripe.org/en/5/contributing/coding_conventions/php_coding_conventions/#unit-testing)

I wouldn't have bothered with this except there are other changes to make here as well, so makes sense to do this at the same time